src/pages/main/components/selected-dataset/build-flow/components/cards/description-card/description-card.tsx

```diff
@@ -0,0 +1,169 @@
+import styles from './description-card.module.css'
+
+import { useEffect, useState } from 'react'
+import { useHistory } from 'react-router'
+import cn from 'classnames'
+
+import { ExploreGenomeTypes } from '@core/enum/explore-genome-types-enum'
+import { getApiUrl } from '@core/get-api-url'
+import { t } from '@i18n'
+import { datasetStore } from '@store/dataset'
+import dirinfoStore from '@store/dirinfo'
+import filterStore from '@store/filter'
+import { Button } from '@ui/button'
+import { Card } from '@ui/card'
+import { Icon } from '@ui/icon'
+import { GlbPagesNames } from '@glb/glb-names'
+import { showToast } from '@utils/notifications/showToast'
+import {
+  exploreCandidateOptionsList,
+  optionsForOpenButton,
+} from '../../wizard/wizard.data'
+import { ICardProps } from '../../wizard/wizard.interface'
+import wizardStore from '../../wizard/wizard.store'
+import { getNextPageData, memorizeLocation } from '../../wizard/wizard.utils'
+import { CardTitleWithEdit } from '../components/card-edit-title'
+import { CardRadioList } from '../components/card-radio-list'
+
+const descriptions: { [key: string]: string } = {}
+let typingTimer: ReturnType<typeof setTimeout>
+
+export const DescriptionCard = (props: ICardProps) => {
+  const {
+    title,
+    id,
+    selectedValue,
+    continueDisabled,
+    contentDisabled,
+    editDisabled,
+  } = props
+  const history = useHistory()
+  const ds = title || datasetStore.datasetName
+
+  const note = descriptions[ds] || dirinfoStore.dirInfoData?.dsDict[ds]?.note
+  const isOpenButton = optionsForOpenButton.includes(selectedValue)
+  const [isEditMode, setEditMode] = useState(false)
+  const [isTyping, setTyping] = useState(false)
+  const [datasetDescription, setDatasetDescription] = useState(
+    note || t('home.datasetDescriptionDefault'),
+  )
+  const fieldDefaultValue =
+    datasetDescription === t('home.datasetDescriptionDefault')
+      ? ''
+      : datasetDescription
+
+  const toggleEditMode = () => {
+    setEditMode(!isEditMode)
+  }
+
+  const openNextPage = () => {
+    const nextPageData = getNextPageData(
+      selectedValue as ExploreGenomeTypes,
+      ds,
+    )
+
+    memorizeLocation(nextPageData.route)
+    history.push(nextPageData.route)
+    filterStore.setMethod(nextPageData.method as GlbPagesNames)
+  }
+
+  const saveDescription = async (description: string) => {
+    const response = await fetch(getApiUrl('dsinfo'), {
```

Comment:
It's restricted to make direct API calls inside tsx components. We make them via service providers and async stores. Here's the manual for async stores: https://github.com/ForomePlatform/Anfisa-React-Client/blob/develop/docs/store/async-data.md
In other words, you must do the following:
1) Go to dataset.provider.ts.
2) Create a new method there that will be making a post request
3) Create an async store in cards/description-card/description-card.async.store.ts
4) Add a fetch method there that will call the service-provider method (you can look up how it's done in other async stores)
5) Create a new instance of it in local description-card.store.ts
6) Call the setQuery method from it with arguments of text to save